java/ql/test/query-tests/security/CWE-297/UnsafeHostnameVerification.java

```diff
@@ -55,12 +56,26 @@ private void functionThatActuallyDisablesVerification() {
 	public void testTrustAllHostnameDependingOnDerivedValue() {
 		String enabled = System.getProperty("disableHostnameVerification");
 		if (Boolean.parseBoolean(enabled)) {
-			HttpsURLConnection.setDefaultHostnameVerifier((hostname, session) -> true);  // GOOD [but detected as BAD].
-			                                                                   // This is GOOD, because it depends on a feature
-			                                                                   // flag, but this is not detected by the query.
+			HttpsURLConnection.setDefaultHostnameVerifier((hostname, session) -> true);  // GOOD, because it depends on a feature
+			                                                                   // flag.
 		}
 	}
 
+	public void testTrustAllHostnameWithExceptions() {
+		HostnameVerifier verifier = new HostnameVerifier() {
+			@Override
+			public boolean verify(String hostname, SSLSession session) {
+				verify(hostname, session.getPeerCertificates());
+				return true; // GOOD [but detected as BAD]. The verification of the certificate is done in another method and
```

Comment:
This kind of scenario is rather rare, of ~100 Alerts I've only seen this pattern once.